### ISSUE TITLE
Compilation errors: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘https’ In file included from networks.c:10:

### ISSUE THREAD

**Issue by JiahuiWangGit** [nanoporetech/flappie#22]
Hi flappie community,

I'm getting compilation errors:

make flappie
mkdir build
cd build && \
	cmake .. -DCMAKE_BUILD_TYPE=Release && \
	make
-- The C compiler identification is GNU 4.4.7
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Looking for hdf5.h
-- Looking for hdf5.h - found
-- Configuring done
-- Generating done
-- Build files have been written to: /projects/li-lab/software/flappie/build
make[1]: Entering directory `/projects/li-lab/software/flappie/build'
make[2]: Entering directory `/projects/li-lab/software/flappie/build'
make[3]: Entering directory `/projects/li-lab/software/flappie/build'
Scanning dependencies of target flappie_objects
make[3]: Leaving directory `/projects/li-lab/software/flappie/build'
make[3]: Entering directory `/projects/li-lab/software/flappie/build'
[  4%] Building C object CMakeFiles/flappie_objects.dir/src/decode.c.o
[  8%] Building C object CMakeFiles/flappie_objects.dir/src/layers.c.o
[ 13%] Building C object CMakeFiles/flappie_objects.dir/src/networks.c.o
In file included from /projects/li-lab/software/flappie/src/networks.c:10:
/projects/li-lab/software/flappie/src/models/flipflop_r941native.h:1: error: expected ‘=’, ‘,’, ‘;’, ‘asm’ or ‘__attribute__’ before ‘https’
In file included from /projects/li-lab/software/flappie/src/networks.c:10:
/projects/li-lab/software/flappie/src/models/flipflop_r941native.h:2:12: error: invalid suffix "dd6adb73a55d95014a6701658d399069b144451dff9179cbefbfb39bb230e" on integer constant
In file included from /projects/li-lab/software/flappie/src/networks.c:11:
/projects/li-lab/software/flappie/src/models/flipflop_r941native5mC.h:2:12: error: invalid suffix "bffdb61b0389174ce2be46ff2e41ae9efaf1332fef6638b8b8e41cf67838" on integer constant
/projects/li-lab/software/flappie/src/networks.c:160: error: ‘_conv_rnnrf_flipflop_r941native_W’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:161: error: ‘_conv_rnnrf_flipflop_r941native_b’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:162: error: ‘conv_rnnrf_flipflop_r941native_stride’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:165: error: ‘_gruB1_rnnrf_flipflop_r941native_iW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:166: error: ‘_gruB1_rnnrf_flipflop_r941native_sW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:167: error: ‘_gruB1_rnnrf_flipflop_r941native_b’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:169: error: ‘_gruF2_rnnrf_flipflop_r941native_iW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:170: error: ‘_gruF2_rnnrf_flipflop_r941native_sW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:171: error: ‘_gruF2_rnnrf_flipflop_r941native_b’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:173: error: ‘_gruB3_rnnrf_flipflop_r941native_iW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:174: error: ‘_gruB3_rnnrf_flipflop_r941native_sW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:175: error: ‘_gruB3_rnnrf_flipflop_r941native_b’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:177: error: ‘_gruF4_rnnrf_flipflop_r941native_iW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:178: error: ‘_gruF4_rnnrf_flipflop_r941native_sW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:179: error: ‘_gruF4_rnnrf_flipflop_r941native_b’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:181: error: ‘_gruB5_rnnrf_flipflop_r941native_iW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:182: error: ‘_gruB5_rnnrf_flipflop_r941native_sW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:183: error: ‘_gruB5_rnnrf_flipflop_r941native_b’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:185: error: ‘_FF_rnnrf_flipflop_r941native_W’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:186: error: ‘_FF_rnnrf_flipflop_r941native_b’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:192: error: ‘_conv_rnnrf_flipflop_r941native5mC_W’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:193: error: ‘_conv_rnnrf_flipflop_r941native5mC_b’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:194: error: ‘conv_rnnrf_flipflop_r941native5mC_stride’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:197: error: ‘_gruB1_rnnrf_flipflop_r941native5mC_iW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:198: error: ‘_gruB1_rnnrf_flipflop_r941native5mC_sW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:199: error: ‘_gruB1_rnnrf_flipflop_r941native5mC_b’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:201: error: ‘_gruF2_rnnrf_flipflop_r941native5mC_iW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:202: error: ‘_gruF2_rnnrf_flipflop_r941native5mC_sW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:203: error: ‘_gruF2_rnnrf_flipflop_r941native5mC_b’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:205: error: ‘_gruB3_rnnrf_flipflop_r941native5mC_iW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:206: error: ‘_gruB3_rnnrf_flipflop_r941native5mC_sW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:207: error: ‘_gruB3_rnnrf_flipflop_r941native5mC_b’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:209: error: ‘_gruF4_rnnrf_flipflop_r941native5mC_iW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:210: error: ‘_gruF4_rnnrf_flipflop_r941native5mC_sW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:211: error: ‘_gruF4_rnnrf_flipflop_r941native5mC_b’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:213: error: ‘_gruB5_rnnrf_flipflop_r941native5mC_iW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:214: error: ‘_gruB5_rnnrf_flipflop_r941native5mC_sW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:215: error: ‘_gruB5_rnnrf_flipflop_r941native5mC_b’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:217: error: ‘_FF_rnnrf_flipflop_r941native5mC_W’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:218: error: ‘_FF_rnnrf_flipflop_r941native5mC_b’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:224: error: ‘_conv_rnnrf_flipflop_r10Cpcr_W’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:225: error: ‘_conv_rnnrf_flipflop_r10Cpcr_b’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:226: error: ‘conv_rnnrf_flipflop_r10Cpcr_stride’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:229: error: ‘_gruB1_rnnrf_flipflop_r10Cpcr_iW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:230: error: ‘_gruB1_rnnrf_flipflop_r10Cpcr_sW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:231: error: ‘_gruB1_rnnrf_flipflop_r10Cpcr_b’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:233: error: ‘_gruF2_rnnrf_flipflop_r10Cpcr_iW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:234: error: ‘_gruF2_rnnrf_flipflop_r10Cpcr_sW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:235: error: ‘_gruF2_rnnrf_flipflop_r10Cpcr_b’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:237: error: ‘_gruB3_rnnrf_flipflop_r10Cpcr_iW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:238: error: ‘_gruB3_rnnrf_flipflop_r10Cpcr_sW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:239: error: ‘_gruB3_rnnrf_flipflop_r10Cpcr_b’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:241: error: ‘_gruF4_rnnrf_flipflop_r10Cpcr_iW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:242: error: ‘_gruF4_rnnrf_flipflop_r10Cpcr_sW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:243: error: ‘_gruF4_rnnrf_flipflop_r10Cpcr_b’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:245: error: ‘_gruB5_rnnrf_flipflop_r10Cpcr_iW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:246: error: ‘_gruB5_rnnrf_flipflop_r10Cpcr_sW’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:247: error: ‘_gruB5_rnnrf_flipflop_r10Cpcr_b’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:249: error: ‘_FF_rnnrf_flipflop_r10Cpcr_W’ undeclared here (not in a function)
/projects/li-lab/software/flappie/src/networks.c:250: error: ‘_FF_rnnrf_flipflop_r10Cpcr_b’ undeclared here (not in a function)
make[3]: *** [CMakeFiles/flappie_objects.dir/src/networks.c.o] Error 1
make[3]: Leaving directory `/projects/li-lab/software/flappie/build'
make[2]: *** [CMakeFiles/flappie_objects.dir/all] Error 2
make[2]: Leaving directory `/projects/li-lab/software/flappie/build'
make[1]: *** [all] Error 2
make[1]: Leaving directory `/projects/li-lab/software/flappie/build'
make: *** [build/flappie] Error 2

**Comment by tmassingham-ont** [comment]
Hello.  This looks like the models have not been checked out properly using `git lfs`.  See https://github.com/nanoporetech/flappie#git-lfs-missing

**Comment by JiahuiWangGit** [comment]
Hi @tmassingham-ont, thanks for reply. It seems like the solution in your link didn't work for my working environment:

[wangj@helix180 flappie]$ curl -s https://packagecloud.io/install/repositories/github/git-lfs/script.deb.sh | bash
Unfortunately, your operating system distribution and version are not supported by this script.

You can override the OS detection by setting os= and dist= prior to running this script.
You can find a list of supported OSes and distributions on our website: https://packagecloud.io/docs#os_distro_version

For example, to force Ubuntu Trusty: os=ubuntu dist=trusty ./script.sh

Please email support@packagecloud.io and let us know if you run into any issues.

**Comment by JiahuiWangGit** [comment]
Hi @tmassingham-ont, I solved the issue above by adding git-lfs in path. It has another error:

[wangj@helix180 flappie]$ make flappie
cd build && \
	cmake .. -DCMAKE_BUILD_TYPE=Release && \
	make
-- The C compiler identification is GNU 4.4.7
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Looking for hdf5.h
-- Looking for hdf5.h - found
-- Configuring done
-- Generating done
-- Build files have been written to: /projects/li-lab/software/flappie/build
make[1]: Entering directory `/projects/li-lab/software/flappie/build'
make[2]: Entering directory `/projects/li-lab/software/flappie/build'
make[3]: Entering directory `/projects/li-lab/software/flappie/build'
Scanning dependencies of target flappie_objects
make[3]: Leaving directory `/projects/li-lab/software/flappie/build'
make[3]: Entering directory `/projects/li-lab/software/flappie/build'
[  4%] Building C object CMakeFiles/flappie_objects.dir/src/decode.c.o
[  8%] Building C object CMakeFiles/flappie_objects.dir/src/layers.c.o
/projects/li-lab/software/flappie/src/layers.c:12:23: error: cblas.h: No such file or directory
/projects/li-lab/software/flappie/src/layers.c: In function ‘convolution’:
/projects/li-lab/software/flappie/src/layers.c:207: warning: implicit declaration of function ‘cblas_sgemv’
/projects/li-lab/software/flappie/src/layers.c:207: error: ‘CblasColMajor’ undeclared (first use in this function)
/projects/li-lab/software/flappie/src/layers.c:207: error: (Each undeclared identifier is reported only once
/projects/li-lab/software/flappie/src/layers.c:207: error: for each function it appears in.)
/projects/li-lab/software/flappie/src/layers.c:207: error: ‘CblasTrans’ undeclared (first use in this function)
/projects/li-lab/software/flappie/src/layers.c:234: warning: implicit declaration of function ‘cblas_sgemm’
/projects/li-lab/software/flappie/src/layers.c:234: error: ‘CblasNoTrans’ undeclared (first use in this function)
/projects/li-lab/software/flappie/src/layers.c: In function ‘gru_step’:
/projects/li-lab/software/flappie/src/layers.c:530: error: ‘CblasColMajor’ undeclared (first use in this function)
/projects/li-lab/software/flappie/src/layers.c:530: error: ‘CblasTrans’ undeclared (first use in this function)
/projects/li-lab/software/flappie/src/layers.c: In function ‘grumod_step’:
/projects/li-lab/software/flappie/src/layers.c:679: error: ‘CblasColMajor’ undeclared (first use in this function)
/projects/li-lab/software/flappie/src/layers.c:679: error: ‘CblasTrans’ undeclared (first use in this function)
/projects/li-lab/software/flappie/src/layers.c: In function ‘gru_relu_step’:
/projects/li-lab/software/flappie/src/layers.c:836: error: ‘CblasColMajor’ undeclared (first use in this function)
/projects/li-lab/software/flappie/src/layers.c:836: error: ‘CblasTrans’ undeclared (first use in this function)
/projects/li-lab/software/flappie/src/layers.c: In function ‘lstm_step’:
/projects/li-lab/software/flappie/src/layers.c:999: error: ‘CblasColMajor’ undeclared (first use in this function)
/projects/li-lab/software/flappie/src/layers.c:999: error: ‘CblasTrans’ undeclared (first use in this function)
make[3]: *** [CMakeFiles/flappie_objects.dir/src/layers.c.o] Error 1
make[3]: Leaving directory `/projects/li-lab/software/flappie/build'
make[2]: *** [CMakeFiles/flappie_objects.dir/all] Error 2
make[2]: Leaving directory `/projects/li-lab/software/flappie/build'
make[1]: *** [all] Error 2
make[1]: Leaving directory `/projects/li-lab/software/flappie/build'
make: *** [build/flappie] Error 2

**Comment by JiahuiWangGit** [comment]
Hi @tmassingham-ont, thanks for reply. I solved the issue above by adding git-lfs in path. There is another issue:
`[wangj@helix180 flappie]$ make flappie
cd build && \
	cmake .. -DCMAKE_BUILD_TYPE=Release && \
	make
-- The C compiler identification is GNU 4.4.7
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Looking for hdf5.h
-- Looking for hdf5.h - found
-- Configuring done
-- Generating done
-- Build files have been written to: /projects/li-lab/software/flappie/build
make[1]: Entering directory `/projects/li-lab/software/flappie/build'
make[2]: Entering directory `/projects/li-lab/software/flappie/build'
make[3]: Entering directory `/projects/li-lab/software/flappie/build'
Scanning dependencies of target flappie_objects
make[3]: Leaving directory `/projects/li-lab/software/flappie/build'
make[3]: Entering directory `/projects/li-lab/software/flappie/build'
[  4%] Building C object CMakeFiles/flappie_objects.dir/src/decode.c.o
[  8%] Building C object CMakeFiles/flappie_objects.dir/src/layers.c.o
/projects/li-lab/software/flappie/src/layers.c:12:23: error: cblas.h: No such file or directory
/projects/li-lab/software/flappie/src/layers.c: In function ‘convolution’:
/projects/li-lab/software/flappie/src/layers.c:207: warning: implicit declaration of function ‘cblas_sgemv’
/projects/li-lab/software/flappie/src/layers.c:207: error: ‘CblasColMajor’ undeclared (first use in this function)
/projects/li-lab/software/flappie/src/layers.c:207: error: (Each undeclared identifier is reported only once
/projects/li-lab/software/flappie/src/layers.c:207: error: for each function it appears in.)
/projects/li-lab/software/flappie/src/layers.c:207: error: ‘CblasTrans’ undeclared (first use in this function)
/projects/li-lab/software/flappie/src/layers.c:234: warning: implicit declaration of function ‘cblas_sgemm’
/projects/li-lab/software/flappie/src/layers.c:234: error: ‘CblasNoTrans’ undeclared (first use in this function)
/projects/li-lab/software/flappie/src/layers.c: In function ‘gru_step’:
/projects/li-lab/software/flappie/src/layers.c:530: error: ‘CblasColMajor’ undeclared (first use in this function)
/projects/li-lab/software/flappie/src/layers.c:530: error: ‘CblasTrans’ undeclared (first use in this function)
/projects/li-lab/software/flappie/src/layers.c: In function ‘grumod_step’:
/projects/li-lab/software/flappie/src/layers.c:679: error: ‘CblasColMajor’ undeclared (first use in this function)
/projects/li-lab/software/flappie/src/layers.c:679: error: ‘CblasTrans’ undeclared (first use in this function)
/projects/li-lab/software/flappie/src/layers.c: In function ‘gru_relu_step’:
/projects/li-lab/software/flappie/src/layers.c:836: error: ‘CblasColMajor’ undeclared (first use in this function)
/projects/li-lab/software/flappie/src/layers.c:836: error: ‘CblasTrans’ undeclared (first use in this function)
/projects/li-lab/software/flappie/src/layers.c: In function ‘lstm_step’:
/projects/li-lab/software/flappie/src/layers.c:999: error: ‘CblasColMajor’ undeclared (first use in this function)
/projects/li-lab/software/flappie/src/layers.c:999: error: ‘CblasTrans’ undeclared (first use in this function)
make[3]: *** [CMakeFiles/flappie_objects.dir/src/layers.c.o] Error 1
make[3]: Leaving directory `/projects/li-lab/software/flappie/build'
make[2]: *** [CMakeFiles/flappie_objects.dir/all] Error 2
make[2]: Leaving directory `/projects/li-lab/software/flappie/build'
make[1]: *** [all] Error 2
make[1]: Leaving directory `/projects/li-lab/software/flappie/build'
make: *** [build/flappie] Error 2
`
